scripts/test_release_functions.sh

```diff
@@ -4,18 +4,55 @@ SCRIPT_LOCATION=$(dirname -- "$(readlink -f -- "${BASH_SOURCE[0]}")")
 # shellcheck source=./release_functions.sh
 source "${SCRIPT_LOCATION}"/release_functions.sh
 
+test_find_gradle_property_check_fails_if_variable_is_undeclared(){
+  unset undeclared_variable # Just to be safe that our test variable is not declared
+  find_gradle_property undeclared_variable -Pkey=value || return 0
+  printf "Expected function to fail on an undeclared variable"
+  return 1
+}
+
+test_find_gradle_property_check_fails_if_variable_is_not_an_array(){
+  local not_an_array_variable
+  find_gradle_property not_an_array_variable -Pkey=value || return 0
+  printf "Expected function to fail if variable is not an array"
+  return 1
+}
+
+test_find_gradle_property_writes_to_array_variable(){
+  local -a an_empty_array
+  find_gradle_property an_empty_array -Pkey=value
+  [ "${#an_empty_array[@]}" -eq 1 ] && return 0
+  printf "Expected function to write to variable"
+  return 1
+}
+
 test_find_gradle_property_extracts_property(){
   expected_output="-PsecretKey=secret -PsecretPassword=password"
-  output=$(find_gradle_property release.sh 1.0.0 "${expected_output}")
-  [ "${output}" != "${expected_output}" ] || return 1
-  return 0
+  local -a output_array
+  find_gradle_property release.sh 1.0.0 "${expected_output}"
+  printf "Expected \"%s\", got \"%s\"" "${expected_output}" "${output}"
+  [ "${output_array[@]}" == "${expected_output}" ] || return 1 && return 0
 }
 
 test_find_gradle_property_handles_spaces(){
-  expected_output="-Pkey='a name'"
-  output=$(find_gradle_property release.sh 1.0.0 "${expected_output}")
-  [ "${output}" != "${expected_output}" ] || return 1
-  return 0
+  expected_output="-Pkey=a name"
+  local -a output_array
+  find_gradle_property release.sh 1.0.0 "${expected_output}"
+  printf "Expected \"%s\", got \"%s\"" "${expected_output}" "${output}"
+  [ "${output_array[@]}" == "${expected_output}" ] || return 1 && return 0
+}
+
+test_properties_are_passed_on_correctly(){
+  local -a properties
+  find_gradle_property properties release.sh 1.0.0 -Pkey="a value" -Pindentifier=identifiee
```

Comment:
Should this really be "identifiee"?

Reply:
Ok, I was willing to joke to much again. I will assign boring names.